Feature 2 add license

nein09 opened the pull request:
Here's my first stab at #2 . Do we want the `licenseText` field in Solr? Also @mbjones was wondering if we could add the license URL to this field if it isn't provided - would that happen in this repo, or would it be elsewhere in the DataONE stack?

mbjones commented:
> Also @mbjones was wondering if we could add the license URL to this field if it isn't provided

Thanks!

I think it would be best to have all three values for each dataset -- license name, license text, and license URL. The last of those is the most machine readable, and so if we can definitely match the specific license URL based on data from the name or text, I would be all for that. I think we should be careful in asserting the url in cases where we are uncertain whether a given text string is a clean match.

I assigned this PR for @taojing2002 to review as he has much better familiarity with the current indexing system than I do.

nein09 commented:
@mbjones When you say "license text", do you mean the full text of the license?

mbjones commented:
Sort of. I meant the text description of the license that people provide in their metadata records -- sometimes there is a paragraph describing usage rights. I don't think this applies to the schema.org case, which shifted to the structured format described in SOSO.

nein09 commented:
Per our conversation in today's meeting, I made the tests use the production solr schema file, and deleted some old test installations of solr that don't seem to be used anywhere.

nein09 commented:
It turns out that changing `solr.schema.path` in `solr.properties` doesn't actually make it use the production path, so I have reverted that change. Also, if I just copy the production schema file to the `solr5home/collection1/conf` directory in the test tree, I get errors about duplicate `funding` fields. There are duplicate `funding` fields in the production schema, but I don't understand why the test infrastructure cares and production doesn't!

So I applied my schema changes to the test schema that is getting used, and deleted the old test installations (`solr`, `solr4home`).  If anyone has any insights here (or thinks I can delete one of the `funding` fields in test and production), I'd love to hear them.

Also, it sounds like @datadavev is the person to look at my schema changes and tell me if they make sense, so I'd appreciate it if you can have a look, Dave.

Thank you!

taojing2002 reviewed (approved):
Looks good to me.
